src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -1007,6 +1008,53 @@ public async Task Alpn_NonH3_NegotiationFailure()
             await new[] { clientTask, serverTask }.WhenAllOrAnyFailed(200_000);
         }
 
+        [Fact]
+        public async Task Alpn_NonH3_FailureEstablishConnection()
+        {
+            var options = new Http3Options() { Alpn = "h3-29" }; // anything other than "h3"
+            using Http3LoopbackServer server = CreateHttp3LoopbackServer(options);
+            using var clientDone = new SemaphoreSlim(0);
+
+            Task serverTask = Task.Run(async () =>
```

Comment:
`serverTask` isn't adding any value here, is it? It is not doing anything beside waiting for client to be done. And you already do that by awaiting `clientTask` below. From what I see, both `serverTask` and `clientDone` semaphore can be safely removed.

But as we don't need to have any logic in parallel with what client is doing, you don't need the client side code to be a task either (as long as there are `WaitAsync`s on each async call, to be sure we would not hang forever on `await`, -- which you have already)